src/lv_widgets/lv_cpicker.c

```diff
@@ -996,23 +999,85 @@ static lv_res_t double_click_reset(lv_obj_t * cpicker)
     return LV_RES_OK;
 }
 
-static lv_color_t angle_to_mode_color(lv_obj_t * cpicker, uint16_t angle)
+#define SWAPPTR(A, B) do { uint8_t * t = A; A = B; B = t; } while(0)
+#define HSV_PTR_SWAP(sextant,r,g,b)     if((sextant) & 2) { SWAPPTR((r), (b)); } if((sextant) & 4) { SWAPPTR((g), (b)); } if(!((sextant) & 6)) { \
+                                                if(!((sextant) & 1)) { SWAPPTR((r), (g)); } } else { if((sextant) & 1) { SWAPPTR((r), (g)); } } 
+
+/* Based on the idea from https://www.vagrearg.org/content/hsvrgb
+   Here we want to compute an approximate RGB value from a HSV input color space. We don't want to be accurate 
+   (for that, there's lv_color_hsv_to_rgb), but we want to be fast.
+   
+   Few tricks are used here: Hue is in range [0; 6 * 256] (so that the sextant is in the high byte and the fractional part is in the low byte)
+   both s and v are in [0; 255] range (very convenient to avoid divisions).
+
+   We fold all symmetry by swapping the R, G, B pointers so that the code is the same for all sextants.
+   We replace division by 255 by a division by 256, a.k.a a shift right by 8 bits. 
+   This is wrong, but since this is only used to compute the pixels on the screen and not the final color, it's ok.
+ */
+static void fast_hsv2rgb(uint16_t h, uint8_t s, uint8_t v, uint8_t *r, uint8_t *g , uint8_t *b);
+static void fast_hsv2rgb(uint16_t h, uint8_t s, uint8_t v, uint8_t *r, uint8_t *g , uint8_t *b)
+{
+    if (!s) { *r = *g = *b = v; return; }
+
+    uint8_t sextant = h >> 8;
+    HSV_PTR_SWAP(sextant, r, g, b); /* Swap pointers so the conversion code is the same */
+
+    *g = v;     
+
+    uint8_t bb = ~s;
+    uint16_t ww = v * bb; /* Don't try to be precise, but instead, be fast */
+    *b = ww >> 8;
+
+    uint8_t h_frac = h & 0xff;
+
+    if(!(sextant & 1)) { 
+        /* Up slope */
+        ww = !h_frac ? ((uint16_t)s << 8) : (s * (uint8_t)(-h_frac)); /* Skip multiply if not required */
+    } else { 
+        /* Down slope */
+        ww = s * h_frac;
+    }
+    bb = ww >> 8;
+    bb = ~bb;
+    ww = v * bb;
+    *r = ww >> 8;
+}
+
+static lv_color_t angle_to_mode_color_fast(lv_obj_t * cpicker, uint16_t angle)
 {
     lv_cpicker_ext_t * ext = lv_obj_get_ext_attr(cpicker);
-    lv_color_t color;
+    uint8_t r = 0, g = 0, b = 0;
+    static uint16_t h = 0;
+    static uint8_t s = 0, v = 0, m = 255;
+    
     switch(ext->color_mode) {
         default:
         case LV_CPICKER_COLOR_MODE_HUE:
-            color = lv_color_hsv_to_rgb(angle % 360, ext->hsv.s, ext->hsv.v);
+            /* Don't recompute costly scaling if it does not change */
+            if (m != ext->color_mode) {
+              s = (uint8_t)(ext->hsv.s * 2.55f); v = (uint8_t)(ext->hsv.v * 2.55f);
```

Comment:
Is it possible to use fixed-point math for these computations? We generally don't allow the use of floats within the library.  Since 2.55 is equal to 51/20 you could try `(ext->hsv.s * 51 / 20)`.

Reply:
Done